### PR TITLE
CFE-2994: cf-check: Respect workdir from config.h

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by olehermanse** [comment]
@olehermanse Please test this with packages from jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2063/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/2063/

**Comment by olehermanse** [comment]
@basvandervlies Do you think this is sufficient?

**Comment by basvandervlies** [comment]
See inline comments. Thanks for effort @olehermanse

**Comment by olehermanse** [comment]
@basvandervlies there are no comments, did you forget to "Submit Review" in GitHub?

File: cf-check/utilities.c [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 Seq *default_lmdb_files()
 {
-    const char *state = "/var/cfengine/state/";
+    const char *state = WORKDIR "/state/";
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply cf-check/utilities.c]: Right, but that definition expands to "default" if it isn't used. What we really want is `GetStateDir()` but that is in libpromises. And libpromises is not a dependency for cf-check.

**Comment by olehermanse** [comment]
@cf-bottom jenkins with exotics again

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2067/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/2067/

**Comment by olehermanse** [comment]
Merged as this improves the situation. JIRA ticket is still open and waiting until I have time to work on this.

**Pull request merged**